handlers/rootfs_image_test.go

```diff
@@ -148,3 +150,183 @@ func TestRootfsReadHeader(t *testing.T) {
 		})
 	}
 }
+
+func TestComposeHeader(t *testing.T) {
+
+	tests := map[string]struct {
+		rfs        *Rootfs
+		args       ComposeHeaderArgs
+		err        error
+		verifyFunc func(args ComposeHeaderArgs)
+	}{
+		"wrong version": {
+			rfs:  NewRootfsInstaller(),
+			args: ComposeHeaderArgs{Version: -1},
+			err:  errors.New("ComposeHeader: rootfs-version 0 not supported"),
+		},
+		"version 1 - no tar writer": {
+			rfs:  NewRootfsV1(""),
+			args: ComposeHeaderArgs{},
+			err:  errors.New("writer: tar-writer is nil"),
+		},
+		"version 2 - no tar writer": {
+			rfs:  NewRootfsV2(""),
+			args: ComposeHeaderArgs{},
+			err:  errors.New("writer: tar-writer is nil"),
+		},
+		"version 3 - no tar writer": {
+			rfs:  NewRootfsV3(""),
+			args: ComposeHeaderArgs{},
+			err:  errors.New("ComposeHeader: Payload: can not tar type-info: arch: Can not write to empty tar-writer"),
+		},
+		"version 1 - success": { // TODO - should this succeed with no update files?
+			rfs: NewRootfsV1(""),
+			args: ComposeHeaderArgs{
+				TarWriter: tar.NewWriter(bytes.NewBuffer(nil)),
+			},
+			err: nil,
+		},
+		"version 3 - success": {
+			rfs: NewRootfsV3(""),
+			args: ComposeHeaderArgs{
+				TarWriter: tar.NewWriter(bytes.NewBuffer(nil)),
+			},
+			err: errors.New("ComposeHeader: Payload: can not tar type-info: arch: Can not write to empty tar-writer"),
+		},
+		"version 3 - Test remove augmented provides": {
+			rfs: NewAugmentedRootfs(NewRootfsV3(""), ""),
+			args: ComposeHeaderArgs{
+				TarWriter:  tar.NewWriter(bytes.NewBuffer(nil)),
+				TypeInfoV3: &artifact.TypeInfoV3{ArtifactProvides: &artifact.TypeInfoProvides{}},
+				Augmented:  true,
+			},
+			verifyFunc: func(args ComposeHeaderArgs) { assert.Nil(t, args.TypeInfoV3) },
+		},
+		"error: metadata not empty": {
+			rfs: NewRootfsV1(""),
+			args: ComposeHeaderArgs{
+				TarWriter: tar.NewWriter(bytes.NewBuffer(nil)),
+				MetaData:  map[string]interface{}{"foo": "bar"},
+			},
+			err: errors.New("MetaData not empty in Rootfs.ComposeHeader. This is a bug in the application."),
+		},
+	}
+
+	for name, test := range tests {
+
+		err := test.rfs.ComposeHeader(&test.args)
+		if err != nil {
+			if test.err == nil {
+				t.Errorf("Test %s failed with an unexpected error: %v", name, err)
+				continue
+			}
+			assert.EqualError(t, err, test.err.Error())
+			if test.verifyFunc != nil {
+				test.verifyFunc(test.args)
+			}
+		}
+	}
+}
+
+func TestNewAugmentedInstance(t *testing.T) {
+	rfs := NewRootfsInstaller()
+	inst, err := rfs.NewAugmentedInstance(NewRootfsV2(""))
```

Comment:
The rest are tests for the setters and getters of the udpates. A fair bit of code. Not really sure how helpful they are though, except increasing the coverage. It did find the nil pointer tar writer panic though.

Reply:
Perhaps not useful right now, but they will be in the future, when we switch rootfs updates over to augmented updates (which was the original plan). For now it's a future investment.